pppwn.py

```diff
@@ -259,7 +259,16 @@ def ipcp_negotiation(self):
                              id=pkt[PPP_IPCP].id,
                              options=pkt[PPP_IPCP].options))
 
-    def ppp_negotation(self, cb=None):
+    def ppp_negotation(self, cb=None, ignore_initial_reqs=False):
+        if ignore_initial_reqs: # Ignore initial requests in order to increase the chances of the exploit working
+            num_reqs_to_ignore = 6 # Tested from 6 to 8 on version 10.50 - all give best results then not ignoring
+            num_ignored_reqs = 0
+            print('[*] Ignoring initial {} PS4 requests..'.format(num_reqs_to_ignore))
+            while num_ignored_reqs < num_reqs_to_ignore:
+                pkt = self.s.recv()
```

Comment:
Are those PADI requests that we're ignoring?

Reply:
Yes.. there is timing issue somewhere but is seems to "alleviate" the problem..

Reply:
Then add this to the loop below that checks for PADI

Reply:
@TheOfficialFloW, done.